pkg/discovery/worker/compliance/schedulable_node_manager.go

```diff
@@ -1,43 +1,76 @@
 package compliance
 
 import (
+	"github.com/turbonomic/kubeturbo/pkg/discovery/repository"
 	"github.com/turbonomic/kubeturbo/pkg/discovery/util"
 	api "k8s.io/api/core/v1"
 )
 
+// Structure to keep track of the schedulable state of the nodes
 type SchedulableNodeManager struct {
-	nodes            map[string]*api.Node
-	schedulableNodes map[string]bool
+	cluster *repository.ClusterSummary
+
+	// Map of node name and boolean indicating its schedulable state
+	schedulableStateMap map[string]bool
+
+	// List of unschedulable nodes by node name
+	unSchedulableNodes []string
 }
 
-func NewSchedulableNodeManager(nodes []*api.Node) *SchedulableNodeManager {
-	nodeMap := make(map[string]*api.Node)
-	for _, node := range nodes {
-		nodeMap[node.Name] = node
+// Build a map for identifying the UIDs of the pods on the unschedulable nodes
+func (manager *SchedulableNodeManager) buildNodeToPodMap() map[string][]string {
```

Comment:
Recommend naming this to `buildNodeToPodUIDMap` to avoid confusion. We already have `manager.cluster.NodeNameToPodMap` in use below.